tests/datasets/sql/test_query.py

```diff
@@ -211,3 +213,21 @@ def test_group_2(rc: RawConfig):
     WHERE ("COUNTRY"."CODE" IS NULL OR "COUNTRY"."CODE" = :CODE_1)
       AND ("COUNTRY"."NAME" IS NULL OR "COUNTRY"."NAME")
     '''
+
+
+def test_explicit_ref(rc: RawConfig):
+    assert _build(rc, '''
+    d | r | b | m | property   | type    | ref           | source     | access
+    example                    |         |               |            |
+      |   |   | Country        |         | id            | COUNTRY    |
+      |   |   |   | id         | integer |               | ID         | open
+      |   |   |   | code       | string  |               | NAME       | open
+      |   |   | City           |         | name          | CITY       |
+      |   |   |   | name       | string  |               | NAME       | open
+      |   |   |   | country    | ref     | Country[code] | COUNTRY_ID | open
+    ''', 'example/City') == '''
+    SELECT
+      "CITY"."NAME",
+      "CITY"."COUNTRY_ID"
```

Comment:
Here, we also need to join `Country` and get `Country.id`, in order to be able to get key for `Country.code` and associate it with `Country.id`.